timely_beliefs/beliefs/classes.py

```diff
@@ -289,37 +289,63 @@ def search_session(
+        event_starts_after: Optional[datetime] = None,
+        event_ends_before: Optional[datetime] = None,
+        beliefs_after: Optional[datetime] = None,
+        beliefs_before: Optional[datetime] = None,
+        event_before: Optional[datetime] = None,  # deprecated
+        event_not_before: Optional[datetime] = None,  # deprecated
+        belief_before: Optional[datetime] = None,  # deprecated
+        belief_not_before: Optional[datetime] = None,  # deprecated
         source: Optional[Union[BeliefSource, List[BeliefSource]]] = None,
     ) -> "BeliefsDataFrame":
         """Search a database session for beliefs about sensor events.
         :param session: the database session to use
         :param sensor: sensor to which the beliefs pertain
-        :param event_before: only return beliefs about events that end before this datetime (inclusive)
-        :param event_not_before: only return beliefs about events that start after this datetime (inclusive)
-        :param belief_before: only return beliefs formed before this datetime (inclusive)
-        :param belief_not_before: only return beliefs formed after this datetime (inclusive)
+        :param event_starts_after: only return beliefs about events that start after this datetime (inclusive)
+        :param event_ends_before: only return beliefs about events that end before this datetime (inclusive)
+        :param beliefs_after: only return beliefs formed after this datetime (inclusive)
+        :param beliefs_before: only return beliefs formed before this datetime (inclusive)
         :param source: only return beliefs formed by the given source or list of sources
         :returns: a multi-index DataFrame with all relevant beliefs
-
-        TODO: rename params for clarity: event_finished_before, event_starts_not_before (or similar), same for beliefs
         """
 
+        # todo: deprecate the 'event_before' argument in favor of 'event_ends_before' (announced v1.4.x)
+        event_ends_before = tb_utils.replace_deprecated_argument(
+            "event_before", event_before, "event_ends_before", event_ends_before
+        )
+        # todo: deprecate the 'event_not_before' argument in favor of 'event_starts_after' (announced v1.4.x)
+        event_starts_after = tb_utils.replace_deprecated_argument(
+            "event_not_before",
+            event_not_before,
+            "event_starts_after",
+            event_starts_after,
+        )
+        # todo: deprecate the 'belief_before' argument in favor of 'beliefs_before' (announced v1.4.x)
+        beliefs_before = tb_utils.replace_deprecated_argument(
+            "belief_before", belief_before, "beliefs_before", beliefs_before
+        )
+        # todo: deprecate the 'belief_not_before' argument in favor of 'beliefs_after' (announced v1.4.x)
+        beliefs_after = tb_utils.replace_deprecated_argument(
+            "belief_not_before", belief_not_before, "beliefs_after", beliefs_after
+        )
```

Comment:
This about sums up this PR. Before merging I'll bump the version and fill it in here.